src/mdns/mdns_mdnssd.cpp

```diff
@@ -1304,7 +1401,18 @@ void PublisherMDnsSd::ServiceInstanceResolution::HandleGetAddrInfoResult(DNSServ
     mInstanceInfo.mTtl = aTtl;
 
 exit:
-    if ((!mInstanceInfo.mAddresses.empty() && !moreComing) || aErrorCode != kDNSServiceErr_NoError)
+    if (IsRetryableError(aErrorCode))
```

Comment:
If the same error persists, will we continue to retry?
Should we add a mechanism to end this cycle (e.g., after a certain number of attempts)?